Sources/TuistLoader/ProjectDescriptionHelpers/CachedProjectDescriptionHelpersBuilderFactory.swift

```diff
@@ -0,0 +1,32 @@
+import Foundation
+import Path
+import TuistSupport
+import XcodeGraph
+
+/// CachedProjectDescriptionHelpersBuilderFactory
+///
+/// is a wrapper on top of `ProjectDescriptionHelpersBuilderFactory` that adds an in-memory cache
+/// to avoid re-creating builder for same cacheDirectory
+// swiftlint:disable:next type_name
+public final class CachedProjectDescriptionHelpersBuilderFactory: ProjectDescriptionHelpersBuilderFactoring {
```

Comment:
Do we actually need an extra level of caching? Isn't the underlying problem we _always_ create a new `ProjectDescriptionsBuilder` [here](https://github.com/tuist/tuist/blob/main/Sources/TuistLoader/Loaders/ManifestLoader.swift#L380)? The `cacheDirectory` is the same across runs, so we could even create a single `ProjectDescriptionHelpersBuilder` in the init of `ManifestLoader` or have a `lazy var` in the `ManifestLoader`.

Reply:
> we could even create a single ProjectDescriptionHelpersBuilder in the init of ManifestLoader or have a lazy var 

This will be problematic, because... function for getting `cacheDirectory` from `cacheDirectoryProviderFactory` throwable:
- `throwing init` will lead to cascading changes in the project
- `lazy var` can not  be throwing